### PR TITLE
[ISR] Swatch Image Fallback

### PR DESCRIPTION
**Pull request by kdelorey**
### Summary
Updated swatch picker to fallback to a drop-down picker if any of the swatch images are unavailable. 

image

### CODE REVIEW

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/screens-with-parts/item-detail/option-components/swatch-product-option-part/swatch-product-option-part.component.html [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
         </div>
     </div>
-
-
+    <caption>{{selectedOptionName}}</caption>
 </div>
-<caption>{{selectedOptionName}}</caption>
\ No newline at end of file
+<ng-template #imageSwatchBackup>
+    <mat-select (selectionChange)="selectOption($event.value)" [value]="screenData.selectedOption">
```
Comment (dankaste) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/screens-with-parts/item-detail/option-components/swatch-product-option-part/swatch-product-option-part.component.html:18]: Should probably have a place holder here like we do for the basic-product-option-part when there is no selection

`<mat-select (selectionChange)="optionSelected($event)" [placeholder]="screenData.optionPlaceholder" [value]="screenData.selectedOption">
    <mat-option *ngFor="let option of screenData.options" 
                [value]="option.id" 
                >{{option.name}}</mat-option>
</mat-select>`

**Review by dankaste — CHANGES_REQUESTED** [review]
Just add the place holder and we are good. Not sure why I didn't think to do it this way.

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/screens-with-parts/item-detail/option-components/swatch-product-option-part/swatch-product-option-part.component.html (hunk 1 above)
Comment (kdelorey) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/screens-with-parts/item-detail/option-components/swatch-product-option-part/swatch-product-option-part.component.html:18]: Good catch, I'll get it added.

Comment (kdelorey) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/screens-with-parts/item-detail/option-components/swatch-product-option-part/swatch-product-option-part.component.html:18]: image

Comment (kdelorey) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/screens-with-parts/item-detail/option-components/swatch-product-option-part/swatch-product-option-part.component.html]: Dependent on JumpMind/commerce#3559 to function correctly.

**Review by dankaste — APPROVED** [review]

**Pull request merged**